container/go/pkg/compat/generateManifest.go

```diff
@@ -0,0 +1,131 @@
+/// Copyright 2015 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//////////////////////////////////////////////////////////////////////
+// Generates manifest based on config and layers.
+
+package compat
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+	"github.com/pkg/errors"
+)
+
+// schemaVersion is the schema version of the docker image manifest to generate.
+const schemaVersion = 2
+
+// GenerateManifest generate a manifest.json metadata by reading a config.json file and layer tarballs from src to dst.
+func GenerateManifest(src, dst string) (v1.Manifest, error) {
+	var configDir string
+	var layersDir []string
+
+	imageRunfiles, err := ioutil.ReadDir(src)
+	if err != nil {
+		return v1.Manifest{}, errors.Wrapf(err, "Error reading legacy image files from %s", src)
+	}
+
+	for _, f := range imageRunfiles {
```

Comment:
Discussed offline how to make this more bullet proof. Basically, ensure we never ignore a file. Either we ignore it explicitly or emit an error. Also, take in the name of the image target and construct the full name of the file to expect, not just if the file "contains" config. This can break if the `container_image` rule that produced the image had `config` in its name.